sherpa/cpp_api/brpc/client.cc

```diff
@@ -0,0 +1,161 @@
+
+  if (cntl->Failed()) {
+    LOG(WARNING) << "Fail to send AsrRequest, " << cntl->ErrorText();
+    return;
+  }
+  if (response->nbest_size()) {
+    LOG(INFO) << "Received response from " << cntl->remote_side()
+      << ": status=[" << response->status()
+      << "], transcript=[" << response->nbest(0).transcript()
+      << "], latency=[" << cntl->latency_us() << "us" << "]";
+  }
+}
+
+static constexpr const char *kUsageMessage = R"(
+Online (streaming) automatic speech recognition RPC client with sherpa.
+
+Usage:
+(1) View help information.
+
+  ./bin/client-client --help
+
+(2) Run client
+
+  ./bin/brpc-client \
+    --server=0.0.0.0:6006 \
+    --wav-file=test.wav
+
+)";
+
+int main(int argc, char* argv[]) {
+  // a Channel represents a communication line to a Server.
+  brpc::Channel channel;
+
+  sherpa::ParseOptions po(kUsageMessage);
+  std::string protocol = "baidu_std";
+  std::string connection_type = "";
+  std::string server = "0.0.0.0:6006";
+  std::string load_balancer = "";
+  int32_t connect_timeout_ms = 3000;
+  int32_t timeout_ms = 5000;
+  int32_t max_retry = 3;
+  std::string wav_file = "";
+  po.Register("protocol", &protocol,
+      "Protocol type, defined in src/brpc/options.proto");
+  po.Register("connection_type", &connection_type,
+      "Connection type in [single, pooled, short]");
+  po.Register("server", &server, "IP Address of server");
+  po.Register("load_balancer", &load_balancer,
+      "The algorithm for load balancing");
+  po.Register("timeout_ms", &timeout_ms, "RPC timeout in milliseconds");
+  po.Register("connect_timeout_ms", &connect_timeout_ms,
+      "RPC connect timeout in milliseconds");
+  po.Register("max_retry", &max_retry,
+      "Max retries(not including the first RPC)");
+  po.Register("wav_file", &wav_file, "Wav file to test");
+  po.Read(argc, argv);
+  if (argc <= 1) {
+    po.PrintUsage();
+    exit(EXIT_FAILURE);
+  }
+
+  brpc::ChannelOptions options;
+  options.protocol = protocol;
+  options.connection_type = connection_type;
+  options.connect_timeout_ms = connect_timeout_ms;
+  options.timeout_ms = timeout_ms;
+  options.max_retry = max_retry;
+  if (channel.Init(server.c_str(),
+        load_balancer.c_str(), &options) != 0) {
+    LOG(ERROR) << "Fail to initialize channel";
+    return -1;
+  }
+
+  // normally, you should not call a Channel directly, but instead construct
+  // a stub Service wrapping it. stub can be shared by all threads as well.
+  sherpa::AsrService_Stub stub(&channel);
+
+  // send a request and wait for the response every 0.32 second.
+  const float sample_rate = 16000;
+  const float frame_time = 0.32;  // second
+  size_t frame_size = sample_rate * frame_time * sizeof(int16_t);
+  std::vector<char> pcm(frame_size);
+  FILE * fp = fopen(wav_file.c_str(), "rb");
+  if (fp) {
+    fseek(fp, 44, SEEK_SET);
+  }
+
+  while (fp &&
+      (frame_size =
+       fread(reinterpret_cast<char*>(pcm.data()), sizeof(char), frame_size, fp))
```

Comment:
```suggestion
       fread(static_cast<void*>(pcm.data()), sizeof(char), frame_size, fp))
```
or use just ` fread(pcm.data(), sizeof(char), frame_size, fp)`

The type of `pcm.data()` is already `char*`.